File: problem13.cpp
#include <string>
class Solution {
public:
    int romanToInt(string s) {
        int sum = 0;
        for(int i = 0; i < s.length(); i++){
            if(getVal(s[i]) < getVal(s[i+1])){
                sum += getVal(s[i+1]) - getVal(s[i]);
                i++;
                continue;
            }
            sum += getVal(s[i]);
        }
        return sum;
    }
    int getVal(char a){
        int sum = 0;
        switch(a){
            case 'I': sum += 1; break;
            case 'V': sum += 5; break;
            case 'X': sum += 10; break;
            case 'L': sum += 50; break;
            case 'C': sum += 100; break;
            case 'D': sum += 500; break;
            case 'M': sum += 1000; break;
            default: sum += INT_MIN;
        }
        return sum;
    }
};

File: Grind75/Week 2/problem13.py
class Solution(object):
    def romanToInt(self, s):
        """
        :type s: str
        :rtype: int
        """
        conv = {
            "I": 1,
            "V": 5,
            "X": 10,
            "L": 50,
            "C": 100,
            "D": 500,
            "M": 1000
        }
        val = 0
        skip = False
        for i in range(0, len(s)):
            if(skip):
                skip = False
                continue
            if(i + 1 < len(s)):
                if(conv[s[i+1]] > conv[s[i]]):
                    num = conv[s[i+1]] - conv[s[i]]
                    val += num
                    skip = True
                else:
                    val += conv[s[i]]
            else:
                val += conv[s[i]]
        return val